plugins/lookup/get_moid.py

```diff
@@ -0,0 +1,579 @@
+            if object_path[-1] != '/':
+                # Save object name
+                object_name = _objects_list.pop(-1)
+            _result = await self.get_objects_moid(dc_moid, _objects_list, object_type, object_name)
+            if not _result:
+                return None
+            elif _result and len(_result) > 1:
+                raise AnsibleError("More than one object available: [%s]."
+                                   % ", ".join(list(item['name'] for item in _result)))
+            try:
+                result = _result[0][object_type]
+            except (TypeError, KeyError, IndexError) as e:
+                raise AnsibleError(to_native(e))
+            return result
+        else:
+            raise AnsibleError('The specified object type does not exist. object_type must be one of [%s].'
+                               % ", ".join(list(INVENTORY.keys())))
+
+    async def _get_cluster_moid(self, cluster_name, folder_moid, dc_moid):
+        _filter = {}
+        _filter["datacenters"] = dc_moid
+        _filter["folders"] = folder_moid
+        if cluster_name:
+            _filter["names"] = cluster_name
+        url = self.build_url('cluster', _filter)
+        return await self.fetch(url)
+
+    async def get_objects_moid(self, dc_moid, objects_list, object_type, object_name=None):
+        _objects = objects_list.copy()
+        folder_moid = None
+        cluster_moid = None
+        rp_moid = None
+        _filter = {}
+        _filter["datacenters"] = dc_moid
+
+        # GET folders MoID
+        _result = await self.get_folder_or_rp_moid(dc_moid, _objects, 'folder')
+        if _result:
+            for folder_info in _result:
+                if folder_info['name'] == objects_list[0]:
+                    del objects_list[0]
+        try:
+            folder_moid = _result[-1]['folder']
+        except (TypeError, KeyError, IndexError) as e:
+            raise AnsibleError(to_native(e))
+
+        if object_type == 'folder':
+            return folder_moid
+
+        if object_type in ('vm', 'host', 'datastore', 'network') and not objects_list:
+            if object_name:
+                _filter['names'] = object_name
+            try:
+                _filter["folders"] = folder_moid
+            except (TypeError, KeyError, IndexError) as e:
+                raise AnsibleError(to_native(e))
+            _url = self.build_url(object_type, _filter)
+            return await self.fetch(_url)
+
+        # GET cluster MoID
+        if objects_list:
+            cluster_name = objects_list[0]
+        else:
+            cluster_name = object_name
+        _result = await self._get_cluster_moid(cluster_name, folder_moid, dc_moid)
+
+        if _result and len(_result) == 1:
+            if objects_list and _result[0]['name'] == objects_list[0]:
+                del objects_list[0]
+            try:
+                cluster_moid = _result[0]['cluster']
```

Comment:
why dont use ``if 'cluster' not in _result[0]`` to avoid exception and properly exit